app/controllers/expenses_controller.rb

```diff
@@ -1,9 +1,11 @@
 class ExpensesController < ApplicationController
-
   def create
-    week = current_user.current_week
-    week.expenses.create(amount: params[:expenses][:amount], manner: params[:expenses][:type])
-    render json: { week: week }
+    current_week = current_user.find_week(0.week)
+    next_week = current_user.find_week(1.week) if params[:expenses][:week] == "true"
+    if params[:expenses][:amount].present?
+      week = next_week || current_week
+      week.expenses.create(amount: params[:expenses][:amount], manner: params[:expenses][:type])
+    end
+    render json: { week: current_week }
   end
```

Comment:
Look up ternary operator and protected methods - also I think the param should really be 'next_week' since thats what we are wanting to act on